src/main/java/org/cryptomator/ui/mainwindow/VaultDetailUnlockedController.java

```diff
@@ -35,6 +54,31 @@ public VaultDetailUnlockedController(ObjectProperty<Vault> vault, FxApplicationW
 		this.vaultStatsBuilder = vaultStatsBuilder;
 	}
 
+	public void initialize() {
+		dropZone.setOnDragEntered(this::handleDragEvent);
+		dropZone.setOnDragOver(this::handleDragEvent);
+		dropZone.setOnDragDropped(this::handleDragEvent);
+		dropZone.setOnDragExited(this::handleDragEvent);
+	}
+
+	private void handleDragEvent(DragEvent event) {
+		if (DragEvent.DRAG_OVER.equals(event.getEventType()) && event.getGestureSource() == null && event.getDragboard().hasFiles()) {
+			draggingUnlockedVaultContentOver.set(event.getDragboard().getFiles().stream().map(File::toPath).anyMatch(this::containsUnlockedVaultContent));
```

Comment:
I think it is better to fail the drop, if at least one file does not belong to the vault.

Also i would show a message ("at least one file does belong to this vault") rather than only show the ones we can.

Reply:
I don't think that it's necessary to add this kind of complexity. Usually, you aren't even able to drag & drop from multiple locations since they can't be in focus simultaneously.